src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -282,8 +282,23 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface **
         ifp->offPremiseServicesReachableIPv4.SetNull();
         ifp->offPremiseServicesReachableIPv6.SetNull();
 
+        CHIP_ERROR error;
         uint8_t addressSize;
-        if (interfaceIterator.GetHardwareAddress(ifp->MacAddress, addressSize, sizeof(ifp->MacAddress)) != CHIP_NO_ERROR)
```

Comment:
How addressSize was populated prior your modification ?

Reply:
`interfaceIterator.GetHardwareAddress` takes it by reference and initializes properly.